crates/ruff_cli/src/commands/format.rs

```diff
@@ -35,41 +44,64 @@ pub(crate) fn format(cli: &FormatArguments, overrides: &Overrides) -> Result<Exi
         return Ok(ExitStatus::Success);
     }
 
-    let result = paths
+    let start = Instant::now();
+    let (results, errors): (Vec<_>, Vec<_>) = paths
```

Comment:
Nit: we could use a `receive` and `sender` channel instead:
* One thread receives the `FormatResult` and builds the `FormatSummary`
* The other threads process the items. 

This would remove the need to write all results to two vectors first.

Reply:
Nice idea, I will try this.